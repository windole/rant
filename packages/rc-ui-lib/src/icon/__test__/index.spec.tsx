import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Icon } from '..';

const IconFont = Icon.createFromIconfontCN('//at.alicdn.com/t/font_2763890_w471tfudy4d.js');

describe('Icon', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  it('should render icon with builtin icon name correctly', () => {
    wrapper = mount(<Icon name="success" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render icon with url name correctly', () => {
    wrapper = mount(<Icon name="https://img.yzcdn.com/icon.jpg" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render default slot correctly', () => {
    wrapper = mount(<Icon name="success">Default Slot</Icon>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should change root tag after using tag prop', () => {
    wrapper = mount(
      <Icon name="success" tag="div">
        Default Slot
      </Icon>,
    );
    expect(wrapper.html()).toEqual(
      '<div class="rc-badge__wrapper van-icon van-icon-success">Default Slot</div>',
    );
  });

  it('should render dot correctly', () => {
    wrapper = mount(<Icon name="success" dot />);
    expect(wrapper.find('.rc-badge').html()).toMatchSnapshot();
  });

  it('should render badge correctly', () => {
    wrapper = mount(<Icon name="success" badge={{ content: 9 }} />);
    expect(wrapper.find('.rc-badge').html()).toMatchSnapshot();
  });

  it('should change icon size when using size prop', () => {
    wrapper = mount(<Icon name="success" size={20} />);
    expect(wrapper.getDOMNode().style.fontSize).toEqual('20px');
  });

  it('should render correctly when createFromIconfontCN', () => {
    wrapper = mount(<IconFont name="cuIcon-classify" color="#f44336" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
