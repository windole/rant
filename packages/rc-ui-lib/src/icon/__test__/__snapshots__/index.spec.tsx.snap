// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render badge correctly 1`] = `"<div class=\\"rc-badge rc-badge--fixed\\">9</div>"`;

exports[`Icon should render customer iconfont correctly 1`] = `"<i class=\\"rc-badge__wrapper van-icon van-icon-cuIcon-classify\\" style=\\"color: rgb(244, 67, 54);\\"><svg width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\"><use xlink:href=\\"#cuIcon-classify\\"></use></svg></i>"`;

exports[`Icon should render default slot correctly 1`] = `"<i class=\\"rc-badge__wrapper van-icon van-icon-success\\">Default Slot</i>"`;

exports[`Icon should render dot correctly 1`] = `"<div class=\\"rc-badge rc-badge--dot rc-badge--fixed\\"></div>"`;

exports[`Icon should render icon with builtin icon name correctly 1`] = `"<i class=\\"rc-badge__wrapper van-icon van-icon-success\\"></i>"`;

exports[`Icon should render icon with url name correctly 1`] = `"<i class=\\"rc-badge__wrapper van-icon\\"><img class=\\"rc-icon__image\\" src=\\"https://img.yzcdn.com/icon.jpg\\" alt=\\"https://img.yzcdn.com/icon.jpg\\"></i>"`;
